### PR TITLE
feat(e2e): init e2e and ex auth-test

### PR DESCRIPTION
**Pull request by TendonT52**
## Change made

- [ ] Bug fixes
- [x] New features
- [ ] Breaking changes

## Describe what you have done

-

### New Features

- Add end to end test

### Fix

-

### Others

-

### CODE REVIEW

File: apps/backend/src/prisma/prisma.service.ts [hunk 1]
```diff
@@ -12,4 +12,22 @@ export class PrismaService extends PrismaClient implements OnModuleInit {
...
+      .filter((name) => name !== '_prisma_migrations')
+      .map((name) => `"public"."${name}"`)
+      .join(', ');
+
+    try {
+      await this.$executeRawUnsafe(`TRUNCATE TABLE ${tables} CASCADE;`);
+    } catch (error) {
+      console.log({ error });
+    }
+  }
```
Comment (PongDev) [inline apps/backend/src/prisma/prisma.service.ts:32]: personally, I think use separate database may be more practical. This code is seem very destructive if miss use.

Comment (PongDev) [reply apps/backend/src/prisma/prisma.service.ts:32]: I think we may start test db from docker compose and run it with inject env to prisma or sth like that.

File: apps/backend/test/jest-e2e.json [hunk 2]
```diff
@@ -1,9 +1,11 @@
 {
   "moduleFileExtensions": ["js", "json", "ts"],
-  "rootDir": ".",
+  "rootDir": "../",
```
Comment (PongDev) [inline apps/backend/test/jest-e2e.json:3]: I am not sure if this will break other unit test.

File: package.json [hunk 3]
```diff
@@ -18,7 +18,12 @@
     "dev:backend": "turbo run build --filter config --filter database --filter types && turbo run dev --filter backend --filter config --filter database",
     "lint": "turbo run lint",
     "format": "prettier --write \"**/*.{ts,tsx,md}\"",
-    "prepare": "husky install"
+    "prepare": "husky install",
+    "build:backend": "turbo run build --filter config --filter database --filter types",
```
Comment (PongDev) [inline package.json:22]: oh, this is good idea, we may also change dev:backend to run build:backend before run dev in the future to reduce duplicated code.

File: apps/backend/src/prisma/prisma.service.ts (hunk 1 above)
Comment (PongDev) [reply apps/backend/src/prisma/prisma.service.ts:32]: also to cleanup, may delete the db with remove of docker volume

**Review by PongDev — CHANGES_REQUESTED** [review]
a little change request, other is LGTM!

**Comment by TendonT52** [comment]
I will use another way to do test 
close this for now 
will major change in e2e soon

**Pull request closed without merging**